woocommerce/woocommerce-android
# [Login] Site discovery post-login, part 1

Part of: #7215 

Please don't merge until the second part #7225 is approved.

### Description
This PR is the first part for implementing the site discovery post-login, it implements the following things:
1. New Fragment for the site discovery.
2. The UI part for the site input.
3. Handling the "Find your site address" click.
4. API call to fetch site info.
5. Post-API handling for Jetpack and WPCom sites, it basically just returns to the site picker.
6. Inline errors for fetch failures.

Remaining tasks:
1. Error state for non-wordpress sites.
2. Error state for non-Jetpack sites + Jetpack installation.

The work done here was first implemented in the PR #7216, but I wasn't satisfied with the quality of the code because of depending on `LoginListener`, then I decided to rewrite everything from scratch.

### Testing instructions
1. Open the app, then sign in with a WordPress.com account that's not connected to any site.
2. In the no-stores error screen, click on "New to woo" button and confirm it works as expected.
4. Check the logs for the tracking of: `site_picker_new_to_woo_tapped`
5. Go back to the app.
6. Click on "Enter a site address"
7. Check the logs for the tracking of: `site_picker_enter_site_address_tapped`.
8. Click on "Find your site address"
9. Confirm the dialog works as expected.
10. Click on "Help" button, and confirm it opens the help screen.
11. Edit the content of the site address.
12. Confirm the Continue button is disabled when the address is not a valid URL.
13. Enter an address for a non-working site.
14. Click on Continue.
15. Confirm an inline error is shown.
16. Enter an address for Jetpack connected site that's connected to another account.
17. Click on Continue.
18. Confirm the account mismatch error is shown.

(Repeat 15-17 for WordPress.com sites, atomic and non-atomic)

### Images/gif
https://user-images.githubusercontent.com/1657201/185418880-9493c48f-543c-4d20-9362-088e6ddcf34a.mov
